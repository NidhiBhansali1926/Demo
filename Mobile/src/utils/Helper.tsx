import { NavigationProp } from '@react-navigation/native';
import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { create } from 'react-test-renderer';

import { store } from '../store/index';

import { AuthParamsList } from '@typing/navigation';

export const renderScreen = (screen: ReactElement) => {
  const myScreen = create(<Provider store={store}>{screen}</Provider>);
  return myScreen;
};

export const getPropsWithNavigation = (props?: unknown, navigationPropExtension?: Partial<NavigationProp<AuthParamsList>>) => ({
  ...props,
  navigation: {
    navigate: jest.fn(),
    goBack: jest.fn(),
    replace: jest.fn(),
    dispatch: jest.fn(),
    getState: jest.fn(() => {}),
    isFocused: () => true,
    addListener: jest.fn().mockImplementation((event, callback) => {
      callback();
      return {
        remove: jest.fn(),
      };
    }),
    ...navigationPropExtension,
  },
});
