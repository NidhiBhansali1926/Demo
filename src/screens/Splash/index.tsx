import { useFonts } from 'expo-font';
import React, { useEffect } from 'react';
import { ActivityIndicator, View } from 'react-native';

import { BG_IMAGE } from '@backgrounds';

import { AppBackground } from '@components';
import { useTheme } from '@hooks';
import { Routes } from '@routes';
import { CustomFonts } from '@src/theme/assets/fonts';
import { LogoSVG } from '@svg';
import { SplashScreenProps } from '@typing/navigation';

const Splash = ({ navigation }: SplashScreenProps) => {
  const { Layout, Gutters, colorThemeVariable } = useTheme();
  const [isFontLoaded] = useFonts(CustomFonts);

  const init = async () => {
    await new Promise((resolve) =>
      setTimeout(() => {
        resolve(true);
      }, 2000),
    );
    navigation.replace(Routes.Auth);
  };

  useEffect(() => {
    if (isFontLoaded) {
      init();
    }
  }, [isFontLoaded]);

  return (
    <AppBackground colors={colorThemeVariable.takeSurveyGradient} overlayColor={colorThemeVariable.takeSurveyOverlay} source={BG_IMAGE.takeSurvey}>
      <View style={[Layout.fill, Layout.center]}>
        <LogoSVG />
        <ActivityIndicator size={'large'} style={[Gutters.largeVMargin]} color={colorThemeVariable.background} />
      </View>
    </AppBackground>
  );
};

export default Splash;
