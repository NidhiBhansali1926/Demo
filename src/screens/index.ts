export { default as SplashScreen } from './Splash';
export { default as TakeSurveyScreen } from './Auth/TakeSurvey';
export { default as ElevateAdventureScreen } from './Auth/ElevateAdventure';
export { default as BoostJourneyScreen } from './Auth/BoostJourney';
export { default as MaximizeExperienceScreen } from './Auth/MaximizeExperience';
export { default as EnterTheGloversScreen } from './Auth/EnterTheGlovers';
