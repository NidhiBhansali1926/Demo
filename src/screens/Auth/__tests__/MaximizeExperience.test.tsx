import React from 'react';
import { MaximizeExperienceScreen } from '@src/screens';
import { getPropsWithNavigation, renderScreen } from '@utils/Helper';

describe('Snapshot Test for Rendering MaximizeExperience Screen Component', () => {
  test('Snapshot of MaximizeExperience Screen with Navigation Props', () => {
    const props = getPropsWithNavigation();
    const maximizeExperienceSnapshot = renderScreen(<MaximizeExperienceScreen {...props} />).toJSON();
    expect(maximizeExperienceSnapshot).toMatchSnapshot();
  });
});
