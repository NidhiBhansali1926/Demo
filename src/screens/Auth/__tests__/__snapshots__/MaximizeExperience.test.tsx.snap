// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test for Rendering MaximizeExperience Screen Component Snapshot of MaximizeExperience Screen with Navigation Props 1`] = `
<AppBackground>
  <RCTScrollView
    bounces={false}
    contentContainerStyle={
      [
        "scrollFillValue",
        {
          "borderTopEndRadius": 32,
          "borderTopStartRadius": 32,
          "paddingHorizontal": 20,
        },
        {
          "backgroundColor": "pink",
        },
      ]
    }
    overScrollMode="never"
    showsVerticalScrollIndicator={false}
    style={
      {
        "marginTop": 35,
      }
    }
  >
    <View>
      <Header
        handleOnPressBackIcon={[Function]}
      />
      <SurveyTwoSVG
        style={
          {
            "marginBottom": 48,
            "marginTop": 53,
          }
        }
      />
      <Text
        style={
          [
            undefined,
            undefined,
            {
              "color": "yellow",
            },
          ]
        }
      >
        maximizeExperience:title
      </Text>
      <Text
        style={
          [
            undefined,
            undefined,
            undefined,
            {
              "marginTop": 4,
            },
            {
              "color": "red",
            },
          ]
        }
      >
        maximizeExperience:description
      </Text>
      <AppButton
        handleClick={[Function]}
        name="maximizeExperience:primaryBtn"
      />
    </View>
  </RCTScrollView>
</AppBackground>
`;
